docfx/doxygen_pages.cc

```diff
@@ -122,8 +122,17 @@ std::vector<TocEntry> PagesToc(pugi::xml_document const& doc) {
   std::transform(nodes.begin(), nodes.end(), result.begin(), [](auto const& i) {
     auto const& page = i.node();
     auto const id = std::string_view{page.attribute("id").as_string()};
-    if (id == "indexpage") return TocEntry{"indexpage.md", "README"};
-    return TocEntry{std::string(id) + ".md", std::string(id)};
+    std::ostringstream title;
+    AppendTitle(title, MarkdownContext{}, page);
+    auto filename =
+        std::string(id == "indexpage" ? std::string_view{"index"} : id) + ".md";
+    return TocEntry{std::string{id}, title.str(), std::move(filename)};
+  });
+  std::sort(result.begin(), result.end(), [](auto const& a, auto const& b) {
+    // Make `index.md` (if present) the first entry.
+    if (a.uid == "indexpage") return true;
+    if (b.uid == "indexpage") return false;
```

Comment:
We should reverse these checks (i.e., have the `b.uid` one come first) so that the comparator is a true "strict weak ordering", where `cmp({.uid = "indexpage"}, {.uid = "indexpage"})` is false.

Reply:
Good catch, fixed.